frontends/web/src/components/alert/Alert.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, h } from 'preact';
+import i18n from '../../i18n/i18n';
+import Dialog from '../dialog/dialog';
+import { Button } from '../forms';
+
+var alerted: (message: string) => void;
+
+interface State {
+    message: string;
+    active: boolean;
+}
+
+interface Alert {
+    button: any;
+}
+
+class Alert extends Component<{}, State> {
+    constructor(props) {
+        super(props);
+        alerted = this.alerted;
+        this.state = {
+            message: '',
+            active: false,
+        };
+    }
+
+    componentDidMount() {
+        document.addEventListener('keydown', this.handleKeyDown);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleKeyDown);
+    }
+
+    handleClose = e => {
+        this.setState({
+            active: false,
+            message: '',
+        });
+    }
+
+    handleKeyDown = e => {
+        if (e.keyCode === 13 && this.state.active) this.setState({ active: false });
```

Comment:
It has nothing to do with TypeScript but I tend to always use curly braces even for one-liners by now.